crates/server/src/domain/quests.rs

```diff
@@ -45,6 +47,13 @@ pub async fn start_quest(
         return Err(QuestError::NotFoundOrInactive);
     }
 
+    let active_quests = db.get_active_user_quest_instances(user_address).await?;
+    for quest in active_quests {
+        if quest.quest_id == quest_id {
+            return Err(QuestError::QuestAlreadyStarted);
+        }
+    }
+
```

Comment:
Is it not worth adding a new function to the DB interface to look for a specific active quest and avoid the for-loop?